src/main/java/com/github/manolo8/darkbot/utils/login/LoginUtils.java

```diff
@@ -112,23 +113,40 @@ private static String getPassword(String username, char[] masterPassword) {
     public static void usernameLogin(LoginData loginData) {
         try {
             usernameLogin(loginData, "www");
-        } catch (Exception e) {
-            usernameLogin(loginData, "lp");
+        }catch (Exception e) {
+            try {
+                usernameLogin(loginData, "lp");
+            } catch (Exception f) {
+                f.printStackTrace();
+            }
         }
     }
 
-    public static void usernameLogin(LoginData loginData, String domain) {
-        String loginUrl = Http.create("https://" + domain + ".darkorbit.com/")
-                .consumeInputStream(LoginUtils::getLoginUrl);
+    private static void usernameLogin(LoginData loginData, String domain) throws IOException {
+        URL url = new URL("https://" + domain + ".darkorbit.com/");
+        String frontPage = IOUtils.read(Http.create(url.toString()).getInputStream());
+
+        Map<String, String> extraPostParams = Collections.emptyMap();
+        CaptchaAPI solver = CaptchaAPI.getInstance();
+        if (solver != null) {
+            try {
+                extraPostParams = solver.solveCaptcha(url, frontPage);
+            } catch (Exception e) {
+                System.out.println("Captcha solver failed to resolve login captcha");
+                e.printStackTrace();
+            }
+        }
 
+        String loginUrl = getLoginUrl(frontPage);
```

Comment:
after this, handle if loginURL == null throw a new exception
Consider making a LoginException that WrongCredentialsException extends, so that we can throw it arround here (and also make sure it's cached in the login form